### PR TITLE
Support CUDA 11.2 and NCCL 2.8

### PR DESCRIPTION
**Pull request by kmaehashi**

### CODE REVIEW

**Comment by emcastillo** [comment]
Seems that we need to add cudnn too

**Comment by emcastillo** [comment]
Jenkins, test this please

**Comment by emcastillo** [comment]
```
00:03:32.964281 STDOUT 2148] | Step 1/22 : ARG base_image |  
-- | -- | --
00:03:32.964290 STDOUT 2148] | Step 2/22 : FROM ${base_image} | 2.6s
00:03:35.576098 STDERR 2148] | manifest for nvidia/cuda:11.2-devel-centos7 not found: manifest unknown: manifest unknown
```

File: dist_config.py [hunk 1]
```diff
@@ -162,13 +162,32 @@
...
+                'nccl_2.8.4-1+cuda11.1_x86_64.txz',
             ],
         },
         'verify_image': 'nvidia/cuda:11.1-runtime-{system}',
         'verify_systems': ['ubuntu18.04'],
         'verify_preloads': ['cudnn'],
     },
+    '11.2': {
+        'name': 'cupy-cuda112',
+        'image': 'nvidia/cuda:11.2-devel-centos7',
```
Comment (kmaehashi) [inline dist_config.py:174]:
```suggestion
        'image': 'nvidia/cuda:11.2.1-devel-centos7',
```

**Comment by kmaehashi** [comment]
pfnCI, test this please.

**Comment by emcastillo** [comment]
```
nvcc fatal   : nvcc cannot find a supported version of Microsoft Visual Studio. Only the versions between 2017 and 2019 (inclusive) are supported! The nvcc flag '-allow-unsupported-compiler' can be used to override this version check; however, using an unsupported host compiler may cause compilation failure or incorrect run time execution. Use at your own risk.
--
```

Linux built complained about no space on device.
Windows fails because of unsupported Visual Studio (I think we were using 2015?)

**Comment by emcastillo** [comment]
Blocked by https://github.com/cupy/cupy-release-tools/pull/66

**Comment by kmaehashi** [comment]
Oops, merging https://github.com/cupy/cupy/pull/4698 closed this PR 😅

**Comment by emcastillo** [comment]
still no space left :(

**Comment by kmaehashi** [comment]
Blocked by: https://github.com/cupy/cupy/pull/4713

**Comment by emcastillo** [comment]
Jenkins, test this please

**Comment by emcastillo** [comment]
Jenkins, test this please

**Comment by emcastillo** [comment]
Jenkins, test this please

**Pull request merged**